src/packages/buskill/__init__.py

```diff
@@ -600,15 +623,18 @@ def triggerWin(self):
 
 		windll.user32.LockWorkStation()
 
+#TODO test on other mac kernel version for sierra - big sur
 	def triggerMac(self):
 		msg = "DEBUG: BusKill lockscreen trigger executing now"
 		print( msg ); logger.info( msg )
-
-		try:
-			subprocess.run( ['pmset', 'displaysleepnow'] )
-		except FileNotFoundError as e:
-			subprocess.run( ['/System/Library/CoreServices/Menu Extras/user.menu/Contents/Resources/CGSession', '-suspend'] )
-
+		if self.KERNEL_VERSION.startswith('17') or self.KERNEL_VERSION.startswith('19'): # High Sierra or Catalina
+			subprocess.run(['/System/Library/CoreServices/Menu Extras/user.menu/Contents/Resources/CGSession', '-suspend']) 
+		#elif self.KERNEL_VERSION.startswith('18'): # Mojave
+			# Find which command works
+		else:
+			msg = "ERROR: Mac Kernel" + self.KERNEL_VERSION + "Unsupported"
+			print( msg ); logger.error(msg)
+# subprocess.run( ['pmset', 'displaysleepnow'] )
```

Comment:
wait, when *does* the `pmset` command work? Wouldn't it be better to just try a bunch of commands as fall-back instead of giving-up before trying?

Reply:
the pmset command had the same issue which linked to the bug on my machines on the other version of macOS.